src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -663,4 +663,11 @@ private Course findCourseFromParticipation(StudentParticipation participation) {
 
         return participationService.findOneWithEagerCourse(participation.getId()).getExercise().getCourse();
     }
+
+    @GetMapping(value = "/participation/{participationId}/submissions")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
```

Comment:
I think we need more access control than that. Not every instructor should be allowed to fetch and submission.

Reply:
I agree, we should check if the user is an instructor of the course of the participation

Reply:
You should also do some checks, e.g. what happens if the participation does not exist?

Reply:
Permission checks are done :)

Reply:
looks good to me